core-api/src/main/java/org/silverpeas/core/persistence/datasource/repository/jpa/AbstractJpaEntityRepository.java

```diff
@@ -444,6 +531,24 @@ private String toJPQLQuery(QueryCriteria criteria) {
     return SilverpeasList.wrap(parameters.applyTo(query).getResultList());
   }
 
+  /**
+   * Gets a stream on the query result.<br/>
+   * Please be careful to always close the stream in order to avoid memory leaks!!!
+   * <pre>
+   *   try(Stream<T> object : streamAllFromQuery(...)) {
+   *     // Performing the treatment
+   *   }
+   * </pre>
+   * TODO use an implementation which performs a real stream operation.<br/>
+   * TODO so, getResultList() method call must disappear.<br/>
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1135)